src/txn/txn_rollback_to_stable.c

```diff
@@ -88,6 +103,34 @@ __rollback_abort_newer_insert(
               session, ins->upd, rollback_timestamp, &stable_update_found);
 }
 
+/*
+ * __rollback_col_add_update --
+ *     Add the provided update to the head of the update list.
+ */
+static inline int
+__rollback_col_add_update(
+  WT_SESSION_IMPL *session, WT_REF *ref, WT_COL *cip, WT_UPDATE *upd, uint64_t recno)
+{
+    WT_CURSOR_BTREE cbt;
+    WT_DECL_RET;
+
+    WT_UNUSED(cip);
+
+    __wt_btcur_init(session, &cbt);
+    __wt_btcur_open(&cbt);
+
+    /* Search the page. */
+    WT_ERR(__wt_col_search(&cbt, recno, ref, true, NULL));
+
+    /* Apply the modification. */
+    WT_ERR(__wt_col_modify(&cbt, recno, NULL, upd, WT_UPDATE_INVALID, true));
+
+    return (ret);
+
+err:
+    return (ret);
```

Comment:
The opened cursor needs to be closed here. Also in case of a failure either free the `upd` in the caller or here.

Reply:
Used __wt_btcur_close to close the cursor, also free'd upd in the caller.